views/js/qtiCreator/widgets/static/img/states/Active.js

```diff
@@ -111,18 +105,59 @@ define([
             $src = widget.$form.find('input[name=src]'),
             $mediaResizer = widget.$form.find('.img-resizer'),
             $mediaSpan = widget.$container;
-
         if($src.val()){
-
             //init data-responsive:
-            if(img.data('responsive') === undefined){
+            if (typeof img.data('responsive') === 'undefined') {
                 if(img.attr('width') && !/[0-9]+%/.test(img.attr('width'))){
                     img.data('responsive', false);
                 }else{
                     img.data('responsive', true);
                 }
             }
 
+            imageUtil.getSize(widget.$original.attr('src'), function(size){
```

Comment:
Do you always restart from the source ? 
It could be useful to write a simple function to load the media, for example ; 

```js
function loadMedia(imgQtiElement, $imgNode) {
    //if the imgQtiElement has already a width, a height, a type, and the node is loaded 
    return Promise.resolve({{
                  $node:  $imgNode,
                    type: imgQtiElement.attr('type'),
                    src:  imgQtiElement.attr('src'),
                    width:  imgQtiElement.attr('width'),
                    height: imgQtiElement.attr('height'),
     });
    //ottherwise, use the imageUtil to load the required data
}

Reply:
Oh and include the initial `responsive` state to it too.

Reply:
And extract the unit too.